addon/-private/functional/modifier.ts

```diff
@@ -13,6 +15,15 @@ import FunctionalModifierManager from './modifier-manager';
 const EAGER_MANAGER = new FunctionalModifierManager({ eager: true });
 const LAZY_MANAGER = new FunctionalModifierManager({ eager: false });
 
+// Dev-only util to make sure we only log the options deprecation once per
+// modifier definition. Disabling the lint because in this case the point is
+// that we don't care about what the internal signature is.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let SEEN_FNS: Set<FunctionalModifierDefinition<any>>;
+if (DEBUG) {
+  SEEN_FNS = new Set();
+}
```

Comment:
I actually think we can get away without doing this trick here. Specifically because modules themselves already cache, and we are emitting all of the new deprecations during the initial module evaluation (because that is when `modifier(...)` is actually called).

In other words, I'm pretty sure that even if you remove this set we cannot have multiple deprecations for the same modifier.

Reply:
Ah! I hadn’t thought of that—thank you!